src/main/java/org/takes/facets/auth/PsBasic.java

```diff
@@ -0,0 +1,175 @@
+        try {
+            final String decoded = new String(
+                DatatypeConverter.parseBase64Binary(
+                    new RqHeaders.Smart(
+                        new RqHeaders.Base(request)
+                    ).single("authorization").split(AUTH_HEAD)[1]
+                )
+            ).trim();
+            final String user = decoded.split(":")[0];
+            if (this.entry.check(
+                user,
+                decoded.substring(user.length() + 1)
+            ).has()) {
+                identity = new Opt.Single<Identity>(
+                    new Identity.Simple(
+                        String.format("urn:basic:%s", user)
+                    )
+                );
+            }
+        } catch (final IOException ex) {
+            identity = new Opt.Empty<Identity>();
+        }
+        return identity;
+    }
+
+    @Override
+    public Response exit(final Response response, final Identity identity)
+        throws IOException {
+        return response;
+    }
+
+    /**
+     * Entry interface that is used to check if the received information is
+     * valid.
+     *
+     * @author Endrigo Antonini (teamed@endrigo.com.br)
+     * @version $Id$
+     * @since 0.20
+     */
+    public interface Entry {
+
+        /**
+         * Check if is a valid user.
+         * @param user User
+         * @param pwd Password
+         * @return Identity.
+         */
+        Opt<Identity> check(String user, String pwd);
+    }
+
+    /**
+     * Entry with fixed credentials.
+     *
+     * @author Endrigo Antonini (teamed@endrigo.com.br)
+     * @version $Id$
+     * @since 0.20
+     *
+     */
+    public static final class Fixed implements PsBasic.Entry {
+
+        /**
+         * Username.
+         */
+        private final transient String username;
+
+        /**
+         * Password.
+         */
+        private final transient String password;
+
+        /**
+         * Ctor.
+         * @param user Username.
+         * @param pwd Password.
+         */
+        public Fixed(final String user, final String pwd) {
+            this.username = user;
+            this.password = pwd;
+        }
+
+        @Override
+        public Opt<Identity> check(final String user, final String pwd) {
+            final Opt<Identity> identity;
+            if (this.username.equals(user) && this.password.equals(pwd)) {
+                identity = new Opt.Single<Identity>(
+                        new Identity.Simple(user)
+                );
+            } else {
+                identity = new Opt.Empty<Identity>();
+            }
+            return identity;
+        }
+    }
+
+    /**
+     * Empty check.
+     *
+     * @author Endrigo Antonini (teamed@endrigo.com.br)
+     * @version $Id$
+     * @since 0.20
+     */
+    public static final class Empty implements PsBasic.Entry {
+
+        @Override
+        public Opt<Identity> check(final String user, final String pwd) {
```

Comment:
@antonini I would call it `enter` instead, since it's not just checking, but "entering"